app.go

```diff
@@ -173,16 +197,11 @@ func (app *App) Run() {
 	}
 }
 
-// Start starts the application.
+// Start executes all the onStart hooks of the fx.Lifecycle.
```

Comment:
Can we explain what this means to a typical service owner, who probably won't ever interact with lifecycle? At the least, we can add a second sentence: "This typically starts long-running goroutines, like network servers or message queue consumers."

Small nit: `onStart` should have initial caps to match the `Hook` field name.